compiler/luci-interpreter/src/kernels/BroadcastTo.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/BroadcastTo.h"
+#include "kernels/Utils.h"
+
+#include "PALBroadcastTo.h"
+
+#include <stdexcept>
+#include <iostream>
+
+namespace luci_interpreter
+{
+
+namespace kernels
+{
+
+// TODO Extract this function to Utils.h
+Shape extractShapeFromTensor(const Tensor *tensor)
+{
+  Shape shape(tensor->shape().num_elements());
+  if (tensor->element_type() == DataType::S32)
+  {
+    const auto *shape_data = tensor->data<int32_t>();
+    for (int i = 0; i < tensor->shape().num_elements(); ++i)
+    {
+      shape.dim(i) = shape_data[i];
+    }
+  }
+  else if (tensor->element_type() == DataType::S64)
+  {
+    const auto *shape_data = tensor->data<int64_t>();
+    for (int i = 0; i < tensor->shape().num_elements(); ++i)
+    {
+      shape.dim(i) = static_cast<int32_t>(shape_data[i]);
+      // Check value overflow
+      LUCI_INTERPRETER_CHECK(static_cast<int64_t>(shape.dim(i)) == shape_data[i]);
+    }
+  }
+  else
+  {
+    LUCI_INTERPRETER_CHECK(false);
+  }
+  return shape;
+}
+
+BroadcastTo::BroadcastTo(const Tensor *input, const Tensor *shape, Tensor *output)
+  : Kernel({input, shape}, {output})
+{
+}
+
+void BroadcastTo::configure()
+{
+  Shape output_shape = extractShapeFromTensor(shape());
+
+  // Check if output shape is broadcastable from input shape
+  int input_rank = input()->shape().num_dims();
+  int output_rank = output_shape.num_dims();
+  int extending_rank = output_rank - input_rank;
+
+  for (int idx = 0; idx < input_rank; ++idx)
+  {
+    LUCI_INTERPRETER_CHECK(input()->shape().dim(idx) == 1 ||
+                           input()->shape().dim(idx) == output_shape.dim(extending_rank + idx));
```

Comment:
can you please explain why add `idx` with `extending_rank` for output_shape?

Reply:
`broadcast_to` operation can stretch the rank (if the rank of the shape is larger than the rank of the input), or the dim value can be stretched. (If the dim value of input shape is 1). 

So, like the [broadcast_to](https://github.com/tensorflow/tensorflow/blob/b487f0246c5804978f86f0a6bf2add1b911af23c/tensorflow/lite/kernels/broadcast_to.cc#L69-L76) of the tflite kernel, codes check whether it is `broadcastable` using `id` and `extending_rank`.

In order to help you understand the above explanations, I attached an example of the result of output shape after operation operation.
1) Stretch rank after operation operation.
2) Stretch dim after operation operation.
```
input shape : (2, 3)
shape shape : (1, 2, 3)
output shape : (1, 2, 3)  -> rank stretch

input shape : (1, 3)
shape shape : (3, 3)
output shape : (3, 3) -> dim stretch
```

Reply:
I don't understand your example with how it's related with `extending_rank`.
Your explanation just jumps to the conclusion without any middle steps.

Reply:
The official document explains BroadcastTo op as follows. -> [link](https://www.tensorflow.org/api_docs/python/tf/broadcast_to)

> Broadcasting is the process of making arrays to have compatible shapes for arithmetic operations. Two shapes are compatible if for each dimension pair they are either equal or one of them is one.

> When broadcasting, if a tensor has fewer axes than necessary its shape is padded on the left with ones. 

Considering this case, it seems that `broadcast_to.cc` in tflite kernel approaches the index with `idx` and `extending_rank` variables. :)